tests/client/test_call_graph.cpp

```diff
@@ -0,0 +1,233 @@
+#include "test_path.hpp"
+#include <client/call_graph.hpp>
+
+#include <gtest/gtest.h>
+
+#include <queue>
+#include <fstream>
+#include <iostream>
+
+using HawkTracer::client::CallGraph;
+
+::testing::AssertionResult SameTree(const std::pair<std::shared_ptr<CallGraph::TreeNode>, int>& expected, 
+                                    const std::pair<std::shared_ptr<CallGraph::TreeNode>, int>& actual)
+{
+    if (expected.second != actual.second)
+    {
+        return ::testing::AssertionFailure() << std::endl
+            << "Expected calls to root: " << expected.second << std::endl
+            << "Actual calls to root: " << actual.second << std::endl;
+    }
+
+    std::queue<std::shared_ptr<CallGraph::TreeNode>> actual_fnc_queue;
+    std::queue<std::shared_ptr<CallGraph::TreeNode>> expected_fnc_queue;
+    actual_fnc_queue.push(actual.first); 
+    expected_fnc_queue.push(expected.first);
+
+    while (!actual_fnc_queue.empty())
+    {
+        auto& actual_fnc = actual_fnc_queue.front();
+        auto& expected_fnc = expected_fnc_queue.front();
+
+        if (expected_fnc->data.label != actual_fnc->data.label)
+        {
+            return ::testing::AssertionFailure() << std::endl 
+                << "Expected label: " << expected_fnc->data.label << std::endl
+                << "Actual label: " << actual_fnc->data.label << std::endl;
+
+        }
+        if (expected_fnc->data.start_ts != actual_fnc->data.start_ts)
+        {
+            return ::testing::AssertionFailure() << std::endl 
+                << "For node with label: " << expected_fnc->data.label << std::endl
+                << "Expected start_ts: " << expected_fnc->data.start_ts << std::endl
+                << "Actual start_ts: " << actual_fnc->data.start_ts << std::endl;
+        }
+        if (expected_fnc->data.stop_ts  != actual_fnc->data.stop_ts)
+        {
+            return ::testing::AssertionFailure() << std::endl 
+                << "For node with label: " << expected_fnc->data.label << std::endl
+                << "Expected stop_ts: " << expected_fnc->data.stop_ts << std::endl
+                << "Actual stop_ts: " << actual_fnc->data.stop_ts << std::endl;
+        }
+        
+        if (expected_fnc->children.size() != actual_fnc->children.size())
+        {
+            return ::testing::AssertionFailure() << std::endl 
+                << "For node with label: " << expected_fnc->data.label << std::endl
+                << "Expected number of children: " << expected_fnc->children.size() << std::endl
+                << "Actual number of children: " << actual_fnc->children.size() << std::endl;
+        }
+
+        for (size_t i = 0; i < actual_fnc->children.size(); ++i)
```

Comment:
using range loop here would simplify the code a little.

Reply:
I'm not sure how to use range loop here since I need to loop over two vectors of the same size: actual_fnc->children and expected_fnc->children